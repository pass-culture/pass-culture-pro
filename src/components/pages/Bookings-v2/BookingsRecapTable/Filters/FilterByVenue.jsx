import React from 'react'
import { ALL_VENUES } from '../utils/filterBookingsRecap'
import PropTypes from 'prop-types'

const FilterByVenue = ({ updateFilters, selectedVenue, venuesFormattedAndOrdered }) => {
  function handleVenueSelection(event) {
    const venueId = event.target.value
    const updatedFilter = { offerVenue: venueId }
    const updatedSelectedContent = { selectedVenue: venueId }
    updateFilters(updatedFilter, updatedSelectedContent)
  }

  return (
    <div className="fw-venues">
      <label
        className="fw-offer-venue-label"
        htmlFor="offer-venue-input"
      >
        {'Lieu'}
      </label>
      <select
        id="offer-venue-input"
        onBlur={handleVenueSelection}
        onChange={handleVenueSelection}
        value={selectedVenue}
      >
        <option value={ALL_VENUES}>
          {'Tous les lieux'}
        </option>
        {venuesFormattedAndOrdered.map(venue => (
          <option
            key={venue.id}
            value={venue.id}
          >
            {venue.displayName}
          </option>
        ))}
      </select>
    </div>
  )
}

FilterByVenue.propTypes = {
  selectedVenue: PropTypes.string.isRequired,
  updateFilters: PropTypes.func.isRequired,
  venuesFormattedAndOrdered: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      displayName: PropTypes.string.isRequired,
    })
  ).isRequired,
}

export default FilterByVenue
