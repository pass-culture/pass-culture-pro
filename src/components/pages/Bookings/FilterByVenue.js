import React, { PureComponent, Fragment } from 'react'
import PropTypes from 'prop-types'

export class FilterByVenue extends PureComponent {
  componentDidMount() {
    this.props.loadVenues()
  }

  onChangeVenue = event => {
    this.props.selectBookingsForVenues(event.target.value)
  }

  render() {
    const { venuesOptions, isDigital } = this.props
    const labelClassName = this.props.isDigital
      ? 'has-text-grey'
      : 'has-text-black'

    return (
      <Fragment>
        <div id="filter-by-venue">
          <label htmlFor="venues" className={labelClassName}>
            {'Choisissez un lieu dans la liste :'}
          </label>
          <select
            id="venues"
            className="pc-selectbox pl24 py5 fs19"
            onChange={this.onChangeVenue}
            disabled={isDigital}>
            <option value="" disabled selected={true}>Choisissez un lieu dans la liste.</option>
            {venuesOptions.map(({ name, id }) => (
              <option key={id} value={id}>
                {name}
              </option>
            ))}
          </select>
        </div>
        <div className="select-digital-offer">
          <div>{'ou :'}</div>
          <input
            id="isDigital"
            className="pc-checkbox input"
            type="checkbox"
            onChange={() => (document.getElementById('venues').value = '')}
            onClick={() => this.props.selectOnlyDigitalVenues(!isDigital)}
            defaultChecked={isDigital}
          />
          <label htmlFor="isDigital">
            {'Cochez cette case pour voir les offres num√©riques'}
          </label>
        </div>
      </Fragment>
    )
  }
}

FilterByVenue.defaultProps = {
  venueId: '',
}

FilterByVenue.propTypes = {
  isDigital: PropTypes.bool.isRequired,
  loadVenues: PropTypes.func.isRequired,
  venueId: PropTypes.string,
  venuesOptions: PropTypes.array.isRequired,
}
