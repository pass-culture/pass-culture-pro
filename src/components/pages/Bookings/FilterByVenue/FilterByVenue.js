import React, { PureComponent, Fragment } from 'react'
import PropTypes from 'prop-types'

class FilterByVenue extends PureComponent {
  componentDidMount() {
    const { loadVenues } = this.props
    loadVenues()
  }

  render() {
    const { venuesOptions, isDigital, selectBookingsForVenues, venueId } = this.props
    const labelClassName = this.props.isDigital
      ? 'has-text-grey'
      : 'has-text-black'

    return (
      <Fragment>
        <div id="filter-by-venue">
          <label htmlFor="venues" className={labelClassName}>
            {'Sélectionnez un lieu :'}
          </label>
          <select
            id="venues"
            className="pc-selectbox pl24 py5 fs19"
            onChange={selectBookingsForVenues}
            value={venueId}
            disabled={isDigital}>
            <option disabled label=" "></option>
            {venuesOptions.map(({ name, id }) => (
              <option key={id} value={id}>
                {name}
              </option>
            ))}
          </select>
        </div>
        <div className="select-digital-offer mt16 mb12">
          <div>{'ou :'}</div>
          <input
            id="isDigital"
            className="pc-checkbox input"
            type="checkbox"
            onClick={() => this.props.selectOnlyDigitalVenues(!isDigital)}
            defaultChecked={isDigital}
          />
          <label htmlFor="isDigital">
            {'Cochez cette case pour voir les offres numériques'}
          </label>
        </div>
      </Fragment>
    )
  }
}

FilterByVenue.defaultProps = {
  venueId: '',
}

FilterByVenue.propTypes = {
  isDigital: PropTypes.bool.isRequired,
  loadVenues: PropTypes.func.isRequired,
  venueId: PropTypes.string,
  venuesOptions: PropTypes.array.isRequired,
}

export default FilterByVenue
