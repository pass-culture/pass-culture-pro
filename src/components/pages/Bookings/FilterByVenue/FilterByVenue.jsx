import React, { PureComponent, Fragment } from 'react'
import PropTypes from 'prop-types'

class FilterByVenue extends PureComponent {
  componentDidMount() {
    const { loadVenues, isUserAdmin, notification, showNotification } = this.props
    if (!notification && isUserAdmin) {
      showNotification()
    }
    loadVenues()
  }

  componentWillUnmount() {
    const { closeNotification, notification } = this.props
    if (notification && notification.tag === 'admin-bookings-access') {
      closeNotification()
    }
  }

  handleOnClick = () => {
    const { isDigital, updateIsFilteredByDigitalVenues } = this.props
    updateIsFilteredByDigitalVenues(!isDigital)
  }

  render() {
    const { isDigital, updateVenueId, venueId, venuesOptions } = this.props
    const labelClassName = isDigital ? 'has-text-grey' : 'has-text-black'

    return (
      <Fragment>
        <div id="filter-by-venue">
          <label
            className={labelClassName}
            htmlFor="venues"
          >
            {'Sélectionner le lieu qui accueille l’offre : '}
          </label>
          <select
            className="pc-selectbox"
            disabled={isDigital}
            id="venues"
            onBlur={updateVenueId}
            onChange={updateVenueId}
            value={venueId}
          >
            <option
              disabled
              label=" - Choisissez un lieu dans la liste - "
              selected
            />
            {venuesOptions.map(({ name, id }) => (
              <option
                key={id}
                value={id}
              >
                {name}
              </option>
            ))}
          </select>
        </div>
        <div className="select-digital-offer mt16 mb12">
          <div>
            {'ou :'}
          </div>
          <input
            className="pc-checkbox input"
            defaultChecked={isDigital}
            id="isDigital"
            onClick={this.handleOnClick}
            type="checkbox"
          />
          <label htmlFor="isDigital">
            {'Cochez cette case pour voir les offres numériques'}
          </label>
        </div>
      </Fragment>
    )
  }
}

FilterByVenue.defaultProps = {
  notification: null,
  venueId: '',
}

FilterByVenue.propTypes = {
  closeNotification: PropTypes.func.isRequired,
  isDigital: PropTypes.bool.isRequired,
  isUserAdmin: PropTypes.bool.isRequired,
  loadVenues: PropTypes.func.isRequired,
  notification: PropTypes.shape(),
  showNotification: PropTypes.func.isRequired,
  updateIsFilteredByDigitalVenues: PropTypes.func.isRequired,
  updateVenueId: PropTypes.func.isRequired,
  venueId: PropTypes.string,
  venuesOptions: PropTypes.arrayOf(PropTypes.shape()).isRequired,
}

export default FilterByVenue
