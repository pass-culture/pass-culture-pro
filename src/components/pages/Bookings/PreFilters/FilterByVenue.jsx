import PropTypes from 'prop-types'
import React from 'react'

import { ALL_VENUES } from './_constants'

const FilterByVenue = ({ updateFilters, selectedVenue, venuesFormattedAndOrdered }) => {
  function handleVenueSelection(event) {
    const venueId = event.target.value
    updateFilters({ offerVenue: venueId })
  }

  return (
    <div className="fw-venues">
      <label
        className="fw-offer-venue-label"
        htmlFor="offer-venue-input"
      >
        {'Lieu'}
      </label>
      <select
        id="offer-venue-input"
        onBlur={handleVenueSelection}
        onChange={handleVenueSelection}
        value={selectedVenue}
      >
        <option value={ALL_VENUES}>
          {'Tous les lieux'}
        </option>
        {venuesFormattedAndOrdered.map(venue => (
          <option
            key={venue.id}
            value={venue.id}
          >
            {venue.displayName}
          </option>
        ))}
      </select>
    </div>
  )
}

FilterByVenue.propTypes = {
  selectedVenue: PropTypes.string.isRequired,
  updateFilters: PropTypes.func.isRequired,
  venuesFormattedAndOrdered: PropTypes.arrayOf(
    PropTypes.shape({
      displayName: PropTypes.string.isRequired,
      id: PropTypes.string.isRequired,
    })
  ).isRequired,
}

export default FilterByVenue
