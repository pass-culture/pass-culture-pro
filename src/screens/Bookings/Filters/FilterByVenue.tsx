import React, { FC } from 'react'

import Select from 'components/layout/inputs/Select'
import { ALL_VENUES_OPTION } from 'constants/booking'

import styles from './Filters.module.scss'

interface IFilterByVenueProps {
  selectedVenueId: string;
  venuesFormattedAndOrdered: SelectOptions;
  updateFilters(filters: Partial<IBookingFilters>): void;
}

const FilterByVenue: FC<IFilterByVenueProps> = ({
  updateFilters,
  selectedVenueId,
  venuesFormattedAndOrdered,
}) => {
  function handleVenueSelection(event: React.ChangeEvent<HTMLSelectElement>): void {
    const venueId = event.target.value
    updateFilters({ offerVenueId: venueId })
  }

  return (
    <Select
      customClass={styles['select-venue']}
      defaultOption={ALL_VENUES_OPTION}
      handleSelection={handleVenueSelection}
      label="Lieu"
      name="lieu"
      options={venuesFormattedAndOrdered}
      selectedValue={selectedVenueId}
    />
  )
}

export default FilterByVenue
