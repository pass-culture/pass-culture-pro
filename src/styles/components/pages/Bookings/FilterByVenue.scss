.select-digital-offer {
  @extend .no-background;

  div {
    margin-top: 16px;
    margin-bottom: rem(12px);
  }

  label {
    white-space: pre;
    margin-left: 0.7rem;
    vertical-align: top;
  }
}

.pc-checkbox {
  background-image: url(/icons/checkbox.svg);
  border: none;
  background-repeat: no-repeat;
  background-size: 20px;
  width: 20px;
  height: 20px;
  padding: 0;
  border-radius: 0;
  margin-bottom: rem(54px);

  &:active,
  &:checked {
    &::after {
      content: '';
      background-repeat: no-repeat;
      background-image: url(/icons/checkbox-check.svg);
      background-position: center;
      position: absolute;
      width: 100%;
      height: 100%;
    }
  }
}

#filter-by-venue {
  .pc-selectbox {

    border-radius: 18px;
    width: 488px;
    height: 36px;
    background-repeat: no-repeat;
    background-position: 455px 14px;
    background-image: url(/icons/dropdown-disclosure-down-s-w.svg);
    background-color: transparent;
    display: block;
    font-size:16px;
    margin: rem(12px) 0 rem(12px);
    box-shadow: 0 2px 6px 0 rgba(0, 0, 0, 0.25);
    -webkit-appearance: none;
    -moz-appearance: none;
  }

  label {
    white-space: pre;
    margin-bottom: rem(12px);
  }
}
