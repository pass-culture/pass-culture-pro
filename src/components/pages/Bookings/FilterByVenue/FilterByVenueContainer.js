import { compose } from 'redux'
import { requestData } from 'redux-saga-data'
import { connect } from 'react-redux'
import withFrenchQueryRouter from '../../../hocs/withFrenchQueryRouter'

import FilterByVenue from './FilterByVenue'
import { closeNotification, showNotification } from 'pass-culture-shared'
import { selectIsUserAdmin } from '../../../../selectors/data/usersSelectors'

import { selectNonVirtualVenues } from '../../../../selectors/data/venuesSelectors'

export const mapDispatchToProps = dispatch => ({
  closeNotification: () => {
    dispatch(closeNotification())
  },
  loadVenues: () => {
    dispatch(
      requestData({
        apiPath: '/venues',
        stateKey: 'venues',
        method: 'GET',
      })
    )
  },
  showNotification: () => {
    dispatch(
      showNotification({
        tag: 'admin-bookings-access',
        text:
          'Votre statut d’administrateur ne permet pas de télécharger le suivi des réservations',
        type: 'info',
      })
    )
  },
  updateIsFilteredByDigitalVenues: payload => {
    dispatch({
      payload,
      type: 'BOOKING_SUMMARY_IS_FILTERED_BY_DIGITAL_VENUES',
    })
  },
  updateVenueId: event => {
    dispatch({
      payload: event.target.value,
      type: 'BOOKING_SUMMARY_UPDATE_VENUE_ID',
    })
  },
})

export const mapStateToProps = state => {
  const { data = {} } = state
  const { venues } = data

  const { bookingSummary = {} } = state
  const { isFilteredByDigitalVenues, venueId } = bookingSummary

  const isUserAdmin = selectIsUserAdmin(state)

  const allVenuesOption = {
    name: 'Tous les lieux',
    id: 'all',
  }

  let venuesOptions = []
  if (!isUserAdmin) {
    const nonVirtualVenues = selectNonVirtualVenues(state, venues)
    venuesOptions = [allVenuesOption, ...nonVirtualVenues]
  }

  return {
    isDigital: isFilteredByDigitalVenues,
    isUserAdmin,
    notification: state.notification,
    venueId,
    venuesOptions,
  }
}

export default compose(
  withFrenchQueryRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(FilterByVenue)
