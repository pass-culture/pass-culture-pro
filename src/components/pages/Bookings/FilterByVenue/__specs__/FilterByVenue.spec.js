import { shallow } from 'enzyme'
import FilterByVenue from '../FilterByVenue'
import React from 'react'

describe('src | components | pages | Bookings | FilterByVenue', () => {
  let props

  beforeEach(() => {
    props = {
      isDigital: false,
      loadVenues: jest.fn(),
      selectBookingsForVenues: jest.fn(),
      selectOnlyDigitalVenues: jest.fn(),
      venuesOptions: [{ name: 'Babar', id: 1 }, { name: 'CÃ©leste', id: 2 }],
    }
  })

  it('should match snapshot', () => {
    // when
    const wrapper = shallow(<FilterByVenue {...props} />)

    // then
    expect(wrapper).toMatchSnapshot()
  })
})
