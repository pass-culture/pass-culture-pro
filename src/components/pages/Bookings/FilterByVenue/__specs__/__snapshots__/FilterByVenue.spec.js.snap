// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | Bookings | FilterByVenue should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterByVenue
    isDigital={false}
    loadVenues={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    updateIsFilteredByDigitalVenues={[MockFunction]}
    updateVenueId={[MockFunction]}
    venueId=""
    venuesOptions={
      Array [
        Object {
          "id": 1,
          "name": "Babar",
        },
        Object {
          "id": 2,
          "name": "Céleste",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          id="filter-by-venue"
        >
          <label
            className="has-text-black"
            htmlFor="venues"
          >
            Sélectionnez un lieu :
          </label>
          <select
            className="pc-selectbox"
            disabled={false}
            id="venues"
            onBlur={[MockFunction]}
            onChange={[MockFunction]}
            value=""
          >
            <option
              disabled={true}
              label=" - Choisissez un lieu - "
              selected={true}
            />
            <option
              value={1}
            >
              Babar
            </option>
            <option
              value={2}
            >
              Céleste
            </option>
          </select>
        </div>,
        <div
          className="select-digital-offer mt16 mb12"
        >
          <div>
            ou :
          </div>
          <input
            className="pc-checkbox input"
            defaultChecked={false}
            id="isDigital"
            onClick={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="isDigital"
          >
            Cochez cette case pour voir les offres numériques
          </label>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="has-text-black"
              htmlFor="venues"
            >
              Sélectionnez un lieu :
            </label>,
            <select
              className="pc-selectbox"
              disabled={false}
              id="venues"
              onBlur={[MockFunction]}
              onChange={[MockFunction]}
              value=""
            >
              <option
                disabled={true}
                label=" - Choisissez un lieu - "
                selected={true}
              />
              <option
                value={1}
              >
                Babar
              </option>
              <option
                value={2}
              >
                Céleste
              </option>
            </select>,
          ],
          "id": "filter-by-venue",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sélectionnez un lieu :",
              "className": "has-text-black",
              "htmlFor": "venues",
            },
            "ref": null,
            "rendered": "Sélectionnez un lieu :",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  disabled={true}
                  label=" - Choisissez un lieu - "
                  selected={true}
                />,
                Array [
                  <option
                    value={1}
                  >
                    Babar
                  </option>,
                  <option
                    value={2}
                  >
                    Céleste
                  </option>,
                ],
              ],
              "className": "pc-selectbox",
              "disabled": false,
              "id": "venues",
              "onBlur": [MockFunction],
              "onChange": [MockFunction],
              "value": "",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "disabled": true,
                  "label": " - Choisissez un lieu - ",
                  "selected": true,
                },
                "ref": null,
                "rendered": null,
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": "Babar",
                  "value": 1,
                },
                "ref": null,
                "rendered": "Babar",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": "Céleste",
                  "value": 2,
                },
                "ref": null,
                "rendered": "Céleste",
                "type": "option",
              },
            ],
            "type": "select",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              ou :
            </div>,
            <input
              className="pc-checkbox input"
              defaultChecked={false}
              id="isDigital"
              onClick={[Function]}
              type="checkbox"
            />,
            <label
              htmlFor="isDigital"
            >
              Cochez cette case pour voir les offres numériques
            </label>,
          ],
          "className": "select-digital-offer mt16 mb12",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ou :",
            },
            "ref": null,
            "rendered": "ou :",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "pc-checkbox input",
              "defaultChecked": false,
              "id": "isDigital",
              "onClick": [Function],
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Cochez cette case pour voir les offres numériques",
              "htmlFor": "isDigital",
            },
            "ref": null,
            "rendered": "Cochez cette case pour voir les offres numériques",
            "type": "label",
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            id="filter-by-venue"
          >
            <label
              className="has-text-black"
              htmlFor="venues"
            >
              Sélectionnez un lieu :
            </label>
            <select
              className="pc-selectbox"
              disabled={false}
              id="venues"
              onBlur={[MockFunction]}
              onChange={[MockFunction]}
              value=""
            >
              <option
                disabled={true}
                label=" - Choisissez un lieu - "
                selected={true}
              />
              <option
                value={1}
              >
                Babar
              </option>
              <option
                value={2}
              >
                Céleste
              </option>
            </select>
          </div>,
          <div
            className="select-digital-offer mt16 mb12"
          >
            <div>
              ou :
            </div>
            <input
              className="pc-checkbox input"
              defaultChecked={false}
              id="isDigital"
              onClick={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="isDigital"
            >
              Cochez cette case pour voir les offres numériques
            </label>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="has-text-black"
                htmlFor="venues"
              >
                Sélectionnez un lieu :
              </label>,
              <select
                className="pc-selectbox"
                disabled={false}
                id="venues"
                onBlur={[MockFunction]}
                onChange={[MockFunction]}
                value=""
              >
                <option
                  disabled={true}
                  label=" - Choisissez un lieu - "
                  selected={true}
                />
                <option
                  value={1}
                >
                  Babar
                </option>
                <option
                  value={2}
                >
                  Céleste
                </option>
              </select>,
            ],
            "id": "filter-by-venue",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sélectionnez un lieu :",
                "className": "has-text-black",
                "htmlFor": "venues",
              },
              "ref": null,
              "rendered": "Sélectionnez un lieu :",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    disabled={true}
                    label=" - Choisissez un lieu - "
                    selected={true}
                  />,
                  Array [
                    <option
                      value={1}
                    >
                      Babar
                    </option>,
                    <option
                      value={2}
                    >
                      Céleste
                    </option>,
                  ],
                ],
                "className": "pc-selectbox",
                "disabled": false,
                "id": "venues",
                "onBlur": [MockFunction],
                "onChange": [MockFunction],
                "value": "",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "disabled": true,
                    "label": " - Choisissez un lieu - ",
                    "selected": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Babar",
                    "value": 1,
                  },
                  "ref": null,
                  "rendered": "Babar",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Céleste",
                    "value": 2,
                  },
                  "ref": null,
                  "rendered": "Céleste",
                  "type": "option",
                },
              ],
              "type": "select",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                ou :
              </div>,
              <input
                className="pc-checkbox input"
                defaultChecked={false}
                id="isDigital"
                onClick={[Function]}
                type="checkbox"
              />,
              <label
                htmlFor="isDigital"
              >
                Cochez cette case pour voir les offres numériques
              </label>,
            ],
            "className": "select-digital-offer mt16 mb12",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ou :",
              },
              "ref": null,
              "rendered": "ou :",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "pc-checkbox input",
                "defaultChecked": false,
                "id": "isDigital",
                "onClick": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Cochez cette case pour voir les offres numériques",
                "htmlFor": "isDigital",
              },
              "ref": null,
              "rendered": "Cochez cette case pour voir les offres numériques",
              "type": "label",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
